src/Render/Options/RenderOptions.php

```
@@ -16,8 +16,27 @@ public function __construct (
 		 */
 		public readonly ?string $levelClass = null,
 		/**
-		 * The locale to translated the labels to
+		 * The locale to translate the labels to
 		 */
 		public readonly ?string $locale = null,
 	) {}
+
+	/**
+	 *
+	 */
+	public static function fromArray (array $data) : self
+	{
+		$filtered = [];
+		$definedProperties = get_object_vars(new self());
+
+		foreach ($data as $key => $value)
+		{
+			if (\array_key_exists($key, $definedProperties))
+			{
+				$filtered[$key] = $value;
+			}
```

Should/Do we need to throw an exception if an unknown property has been passed, as it may hide a potential bug? Or do we not care about that scenario?

---

We actually use that scenario, as we merge `RenderOptions` and `RouteTreeOptions` in the twig function.

Also accepting additional properties is great for forward compatibility.